api/services/auth.js

```diff
@@ -128,6 +129,68 @@ async function registration({ firstName, lastName, email, password, token }) {
   });
 }
 
+/**
+ * TODO: It needs to think about creating link for reset password
+ * @param {*} param0 - object which contents email field
+ */
+
+const resetPasswordRequest = async ({ email }) => {
+  try {
+    let v = await db.Users.findOne({ where: { email } });
+    let linkId = v && (await hashHelpers.createHash(email));
+
+    let forgotPassword =
+      v &&
+      (await db.ForgotPassword.findOrCreate({
+        where: { linkId },
+        defaults: { UserId: v.id },
+      }));
+    let linkId2 = encodeURIComponent(linkId);
+    const info =
+      v && linkId && forgotPassword
+        ? await emailHelpers.sendEmail(
+            email,
+            `Follow link http://localhost:3000/reset?id=${linkId2}`,
+            `http://localhost:3000/reset?id=${linkId2}`
+          )
+        : { message: 'User not founded', status: 400 };
```

Comment:
For errors dont use default object. We have middleware for it. There are a few custom errors classes, so you can write your own error and throw it

Reply:
Done